engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/number/decimal/IntNode.java

```diff
@@ -0,0 +1,31 @@
+package org.enso.interpreter.node.expression.builtin.number.decimal;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.nodes.Node;
+import com.oracle.truffle.api.profiles.ConditionProfile;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.node.expression.builtin.number.utils.BigIntegerOps;
+import org.enso.interpreter.runtime.number.EnsoBigInteger;
+
+@BuiltinMethod(
+    type = "Decimal",
+    name = "int",
+    description = "Truncate a floating-point number to an integer by dropping the fractional part.")
+public class IntNode extends Node {
+  private final ConditionProfile fitsProfile = ConditionProfile.createCountingProfile();
+
+  Object execute(double self) {
```

Comment:
This is strange. If `self` can be converted to `double`, then it is not `EnsoBigInteger`.

Reply:
I'm not sure I understand. self is always a double, never an EnsoBigInteger. The result is either a small or big integer.

Reply:
It's actually been renamed to TruncateNode now, and implements Enso Decimal.truncate. I assume this is the right naming convention.